pkgs/by-name/pn/pnfft/package.nix

```diff
@@ -0,0 +1,59 @@
+{ autoconf
+, automake
+, fetchurl
+, fftwMpi
+, gsl
+, lib
+, libtool
+, llvmPackages
+, pfft
+, precision ? "double"
+, stdenv
+}:
+
+assert lib.elem precision [ "single" "double" "long-double" ];
+
+let
+  fftw' = fftwMpi.override { inherit precision; };
+  pfft' = pfft.override { inherit precision; };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "pnfft-${precision}";
+  version = "1.0.7-alpha";
+
+  src = fetchurl {
+    url = "https://www-user.tu-chemnitz.de/~potts/workgroup/pippig/software/pnfft-${finalAttrs.version}.tar.gz";
+    hash = "sha256-/aVY/1fuMRl1Q2O7bmc5M4aA0taGD+fcQgCdhVYr1no=";
+  };
+
+  outputs = [ "out" "dev" ];
+
+  nativeBuildInputs = [
+    autoconf
+    automake
+    libtool
+  ];
+
+  preConfigure = ''
+    export FCFLAGS="-I${lib.getDev fftw'}/include -I${lib.getDev pfft'}/include"
```

Comment:
the include paths should be picked up automatically, when you put `fftw'` and `pfft'` in `buildInputs`?

Reply:
There is no `NIX_FCFLAGS` that is automatically picked up and the `configure` script doesn't search for `fftw3-mpi.f03` or `pfft.f03`.

Reply:
The compiler wrapper for gfortran should add the `-I` options automatically from the packages in `buildInputs`. I am not sure why this does not work in this case.

Reply:
Maybe because it uses `mpif90` instead of `gfortran`? I'm only guessing.